### DISCUSSION TITLE
Replace all spaces in the filenames with underscores like in v1.X?

### DISCUSSION THREAD

**Discussion by parasiteoflife** [Serene-Arc/bulk-downloader-for-reddit#382] [Q&A] [answered]
All the files I've donwloaded before with BDFR v1.x have no spaces, instead the have underscores, now with the new version the TITLE is saved with spaces, how can I change that back to underscores? The application is redownloading files I already have.

Before it was saving like `RedditorUsername_Title_with_underscores_instead_of_spaces_XXXXXX`
Now is saving it like `RedditorUsername_Title with spaces instead of underscores_YYYYYY`

**Comment by Serene-Arc** [comment, marked as answer]
No, sorry that currently isn't possible. The BDFR downloads the file with minimal changes to the filename at the current time, only doing the minimum necessary to comply with filesystem limitations. If you use the `--search-existing` and `--no-dupes` , then it should prevent any redownloading.

**Reply by parasiteoflife** [reply]
Thanks for the info. It seems I will have to use those two commands. I'm surprised nobody has asked about this, I guess everybody is use those commands already.

One question, where BDFR stores the hashes? Can I specify a path?

**Reply by Serene-Arc** [reply]
It doesn't store them, the BDFR recalculates them on every run to ensure that its idea of the data is correct and current.

**Comment by aliparlakci** [comment]
A better solution is probably to create a file with the list of IDs that already exists with a simple script and provide it to BDfR for every run. Since the newly downloaded files will already be recognized by v2, you don't have to update that list after each run.

**Reply by ymgenesis** [reply]
Is this currently possible in V2? To save downloaded IDs to a file, and have it read the file on each run to skip matching submissions?

**Reply by aliparlakci** [reply]
Totally. See [`--exclude-id-file`](https://github.com/aliparlakci/bulk-downloader-for-reddit#downloader-options) We will modify the name of the options to be more consistent. `--exclude-id-file` will be `--skip-id-file`

**Reply by ymgenesis** [reply]
Is there a way bdfr can generate a list of IDs for all new downloads? That way we don't have to create a script to generate a list of IDs for it to check with --skip-id-file later. Can't seem to find this in the readme.

**Reply by aliparlakci** [reply]
No, there is no such way. Because we don't want to bloat it with the features that can be achieved with helper scripts or chaining commands back to back. BDFR is a simple tool and we want it to stay simple. New functionality is implemented if it is impossible to do with the current state of the project or there needs to be a good reason to have it in the source code. We want bdfr to be flexible enough for users to customize and adapt it to their workflows.

**Reply by ymgenesis** [reply]
Sounds great!
